### PR TITLE
[Mobile Payments] Add extra info in the tracking of payments flow completion

### LINKED ISSUE #9570: Add extra tracking
```
Track amount property with all payment_methods.
Add amount_normalized property (it’d contain only the number and would use a single locale).
Add currency property (currency in the USD/GBP/... format).
```

Context - pdND4G-fv-p2

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #9570

Partially: #9578

### Description

The PR adds `amount_normalized` and `currency` properties to all payment method tracking of the `payments_flow_completed` event. `amount_normalized` means tracking the value in Smallest Currency Unit

### Testing instructions

* Start payments flow, either from order or Payments -> Collect payment
* Run all four payments methods: cash, card, scan to pay, payment link sharing
* Notice in Logcat for every `payments_flow_completed` event `amount_normalized` and `currency` properties

```
Tracked: payments_flow_completed, Properties: {"payment_method":"cash","order_id":2448,"amount":"$10.86","amount_normalized":1086,"currency":"USD","flow":"simple_payment","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","is_debug":true}

Tracked: payments_flow_completed, Properties: {"payment_method":"scan_to_pay","order_id":2449,"amount":"$1,219.86","amount_normalized":121986,"currency":"USD","flow":"simple_payment","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","is_debug":true}

Tracked: payments_flow_completed, Properties: {"payment_method":"card","order_id":2450,"amount":"$1.21","amount_normalized":121,"currency":"USD","flow":"simple_payment","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","is_debug":true}

Tracked: payments_flow_completed, Properties: {"payment_method":"payment_link","order_id":2443,"amount":"$20.00","amount_normalized":2000,"currency":"USD","flow":"order_payment","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","is_debug":true}
```

Try all of this on the 0 or 3 decimals currencies, e.g., KWD and JPY

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by kidinov** [comment]
@samiuelson I want to fix currency conversion here as well as per Cesar's code, so no need to review it till the next week

p1691749124542089/1691737521.977669-slack-C025A8VV728

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt [hunk 1]
```diff
@@ -30,30 +30,108 @@ class PaymentUtilsTest : CardReaderBaseUnitTest() {
     }
 
     @Test
-    fun `when amount $1, then it gets converted to 100 cents`() = testBlocking {
-        val result = paymentUtils.convertBigDecimalInDollarsToLongInCents(BigDecimal(1))
+    fun `whenAmount$1ThenItGetsConvertedTo100Cents`() {
```
Comment (samiuelson) [inline libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: What's the reason for removing the spaces from the method name?

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt [hunk 2]
```diff
@@ -30,30 +30,108 @@ class PaymentUtilsTest : CardReaderBaseUnitTest() {
...
+            TWO_DECIMAL_CURRENCY_CODE,
+        )
 
         assertThat(result).isEqualTo(100)
     }
 
     @Test
-    fun `when amount $1 005cents, then it gets rounded down to 100 cents`() = testBlocking {
-        val result = paymentUtils.convertBigDecimalInDollarsToLongInCents(BigDecimal(1.005))
+    fun `when amount $100 5cents, then it gets rounded down to 100 cents`() {
```
Comment (samiuelson) [inline libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: I think the previous method name was more accurate (005 cents vs 5 cents). WDUT?

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt [hunk 3]
```diff
@@ -30,30 +30,108 @@ class PaymentUtilsTest : CardReaderBaseUnitTest() {
...
+            TWO_DECIMAL_CURRENCY_CODE,
+        )
 
         assertThat(result).isEqualTo(100)
     }
 
     @Test
-    fun `when amount $1 006cents, then it gets rounded up to 101 cents`() = testBlocking {
-        val result = paymentUtils.convertBigDecimalInDollarsToLongInCents(BigDecimal(1.006))
+    fun `when amount $100 6 cents, then gets rounded up to 101 cents`() {
```
Comment (samiuelson) [inline libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: If the currency is two decimal the amount of 1.006 contains 0.6 of cent - assuming 1 cent = 1/100 🤔

Comment (samiuelson) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: Maybe we could add `given two decimal currency, when amount ...` at the beginning of these test methods?

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt [hunk 4]
```diff
@@ -35,18 +36,18 @@ internal class CreatePaymentActionTest : CardReaderBaseUnitTest() {
     private lateinit var action: CreatePaymentAction
     private val paymentIntentParametersFactory = mock<PaymentIntentParametersFactory>()
     private val terminal: TerminalWrapper = mock()
-    private val paymentUtils: PaymentUtils = PaymentUtils()
     private val intentParametersBuilder = mock<PaymentIntentParameters.Builder>()
     private val cardReaderConfigFactory: CardReaderConfigFactory = mock()
+    private val paymentUtils: PaymentUtils = mock()
```
Comment (samiuelson) [inline libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt:41]: Why mocking? 
```suggestion
    private val paymentUtils: PaymentUtils = PaymentUtils
```

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt [hunk 5]
```diff
@@ -30,30 +30,108 @@ class PaymentUtilsTest : CardReaderBaseUnitTest() {
...
     @Test
-    fun `when amount $1 005cents, then it gets rounded down to 100 cents`() = testBlocking {
-        val result = paymentUtils.convertBigDecimalInDollarsToLongInCents(BigDecimal(1.005))
+    fun `when amount $100 5cents, then it gets rounded down to 100 cents`() {
+        val result = paymentUtils.convertToSmallestCurrencyUnit(
+            BigDecimal(1.005),
+            TWO_DECIMAL_CURRENCY_CODE,
+        )
 
         assertThat(result).isEqualTo(100)
```
Comment (samiuelson) [inline libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt:51]: ❓ This test's logic is misleading. In fact, the result of converting the 1.005 value should be 101 - since we apply `.setScale(0, RoundingMode.HALF_UP)` internally. The only reason the result ends up being 100 is caused by the fact the BigDecimal instantiated with double `BigDecimal(1.005)` loses precision (in fact it holds value like `1.00499999999999989341858963598497211933135986328125`). If you instantiate it with string instead - `BigDecimal("1.005")` the result is equal to 101 as (should be) expected.

Comment (samiuelson) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt:51]: 💡I found more insights [here](https://rules.sonarsource.com/java/RSPEC-2111/). 
**TLDR:** 

> **"BigDecimal(double)" should not be used, instead use BigDecimal.valueOf, which uses a string under the covers to eliminate floating point rounding errors, or the constructor that takes a String argument.**

**Review by samiuelson — APPROVED** [review]
Looks good overall 👍🏼  I've added a few improvement ideas. Let me know what you think!

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt (hunk 1 above)
Comment (kidinov) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: I used the android version of currency first, so I had to move it to android tests. Forgot to change it back

Fixed c88d2642ff506e6fecfe95cd979704cae7591001

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt (hunk 2 above)
Comment (kidinov) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: Fixed c88d2642ff506e6fecfe95cd979704cae7591001

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt (hunk 3 above)
Comment (kidinov) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt]: Good point. Fixed c88d2642ff506e6fecfe95cd979704cae7591001

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt (hunk 4 above)
Comment (kidinov) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt:41]: That's a unit test; shouldn't it be tested in isolation? Changes in the `PaymentUtils` will cause this method to fail if we'll call the real code, which may be good and bad at the same time. Wdyt?

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt (hunk 5 above)
Comment (kidinov) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/PaymentUtilsTest.kt:51]: I changed that to the usage of the string, to avoid confusion. eac8d0d6f875ef3dc6d7ef035fba03e29e252966

File: libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt (hunk 4 above)
Comment (samiuelson) [reply libs/cardreader/src/test/java/com/woocommerce/android/cardreader/internal/payments/actions/CreatePaymentActionTest.kt:41]: I see, yes, it makes sense to mock it here, since we have separate tests that test PaymentUtils class
